File: solutions/ruby/matrix/1/matrix.rb
class Matrix
  def initialize(string)
    @matrix = string.lines.map{|row| row.split(' ').map(&:to_i)}
  end

  def row(n)
    @matrix[n-1]
  end

  def column(n)
    @matrix.map{|row| row[n-1]}
  end
end

File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string: str):
        *self._matrix, = map(lambda line: [*map(int,
                             line.split())], matrix_string.split('\n'))

    def row(self, index: int) -> list[int]:
        return self._matrix[index - 1]

    def column(self, index: int) -> list[int]:
        return list(list(zip(*self._matrix))[index - 1])
